runtime/shared_common/ClasspathManagerImpl2.cpp

```diff
@@ -1099,7 +1114,26 @@ SH_ClasspathManagerImpl2::validate(J9VMThread* currentThread, ROMClassWrapper* f
 	if (localFoundAtIndex == -1) {
 		/* We didn't find an exact classpath match. Walk the classpaths and check for validity. Note that the timestamp check is incorporated for efficiency. */
 		localFoundAtIndex = localValidate_CheckAndTimestampManually(currentThread, testCP, testCPIndex, compareTo, foundIdentified, &addToIdentified, staleItem);
+		/* If the classpath has just been identified by localValidate_CheckAndTimestampManually, store it in the identified array */
+		if (addToIdentified && (compareTo->getType()==CP_TYPE_CLASSPATH) && (*_runtimeFlagsPtr & J9SHR_RUNTIMEFLAG_ENABLE_LOCAL_CACHEING)) {
+			/* Identification succeeded. Store identified classpath. */
+			if (local_StoreIdentified(currentThread, compareTo, testCP)==-1) {
```

Comment:
Since the condition tested to get here isn't the same as the condition where identifiedMutex is entered further above, it seems possible for the assertion in local_StoreIdentified() to occur. Shouldn't identifiedMutex be entered if it's not already?

Reply:
Yes, it should enter identifiedMutex if it has not. I thought about this case yesterday in the morning, but after lunch, I forgot.  Fixed now.